backend/flask/app.py

```diff
@@ -124,8 +125,15 @@ def post_req_handler():
         user = User(email=data['email'], password=encrypted_password)
         db.session.add(user)
         db.session.commit()
-        return jsonify({'message': 'Data saved!', 'userId': user.id}), 201
         # return the users id so they can be later identified
+        return jsonify({'message': 'Data saved!', 'userId': user.id}), 201
+    except IntegrityError as e:
+        db.session.rollback()
+        # Check if the error is due to a duplicate email
+        if 'duplicate key value violates unique constraint' in str(e):
```

Comment:
Does IntegrityError have a code / something other than string matching you can use for this check? No worries if not.